src/include/btree_inline.h

```diff
@@ -1093,12 +1151,34 @@ __wt_row_leaf_key(
     copy = WT_ROW_KEY_COPY(rip);
 
     /*
-     * All we handle here are on-page keys (which should be a common case), and instantiated keys
-     * (which start out rare, but become more common as a leaf page is searched, instantiating
-     * prefix-compressed keys).
+     * Handle keys taken directly from the disk image (which should be a common case), instantiated
+     * keys (rare initially, but possibly more common as leaf page search instantiates keys), and
+     * keys built using the most-used page key prefix.
+     *
+     * The most-used page key prefix: the longest group of compressed key prefixes on the page that
+     * can be built from a single, fully instantiated key on the page, was tracked when the page was
+     * read. Build keys in that group by appending the key's bytes to the root key from which it was
+     * compressed.
      */
-    if (__wt_row_leaf_key_info(page, copy, NULL, NULL, &key->data, &key->size))
+    __wt_row_leaf_key_info(page, copy, NULL, &cell, &key_data, &key_size, &key_prefix);
+    if (key_data != NULL && key_prefix == 0) {
+        key->data = key_data;
+        key->size = key_size;
         return (0);
+    }
+    slot = WT_ROW_SLOT(page, rip);
+    if (key_data != NULL && slot > page->prefix_start && slot <= page->prefix_stop) {
+        /* The row-store key can change underfoot; explicitly take a copy. */
+        copy = WT_ROW_KEY_COPY(&page->pg_row[page->prefix_start]);
```

Comment:
This looks essentially identical to `wt_row_leaf_key_work` above. Can/Should it be factored out so there's only one copy?

Reply:
I don't understand what you're saying here, can you please explain further?

Reply:
The code in `btree/row_key.c:342` is almost identical to this code here 1175-1182:
```
copy = WT_ROW_KEY_COPY()
wt_row_leaf_key_info(...)
if (group_key != NULL) {
    wt_buf_grow(...)
    memcpy(...)
    memcpy(...)
...
}
```
If anyone who's not you changes one of these locations for some reason they may not find the other location. So maybe we could refactor so there is only one version of this code? (I realize my use of `only one copy` when you have a variable called `copy` might have been confusing. I meant one copy of the same lines of code.)

Reply:
@sueloverso, I looked at that same chunk myself and it was abstracted out at one point -- all I can say is that my recollection was it was messy and I eventually broke it apart into two separate chunks of code. If you feel strongly, let me know.